### ISSUE TITLE
Offchain workers code shouldn't be on chain

### ISSUE THREAD

**Issue by tomaka** [paritytech/substrate#3290]
Since it doesn't participate in consensus, the code run by the offchain worker shouldn't be included in the `:code` key.

**Comment by andresilva** [comment]
The advantage I see is allowing off-chain code updates the same way we do for the runtime. We can't trust the code but it's useful for honest nodes I guess.

**Comment by kianenigma** [comment]
> The advantage I see is allowing off-chain code updates the same way we do for the runtime. We can't trust the code but it's useful for honest nodes I guess.

Yes, and I think it makes more sense for the this variant of an off-chain program (aka. `fn offchain_worker()`) **to be upgradable with the runtime** and tightly coupled with it. If you don't want that, I don't see anything keeping you from writing your own program that runs next to the client and does whatever it wants, reads stuff and submits back results, both through RPC. Maybe someday the abstractions provided by the runtime offchain worker become so noticeably better than manual scaffolding a new program that then we could re-consider.
